### PR TITLE
DPLY-40 Use DbSessionFactory and DbSession to bridge the Hibernate 3 vs. 4 differences

### PR DESCRIPTION
**Pull request by asp78**
https://issues.openmrs.org/browse/DPLY-40

### CODE REVIEW

**Comment by dkayiwa** [comment]
@asp78 did you get a chance to take a look at this? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java [hunk 1]
```diff
@@ -189,7 +191,7 @@ public void getForm_shouldThrowExceptionForNonExistent() {
 	@Test
 	public void getLocation_shouldFetchByUuid() {
 		Location unknown = Context.getLocationService().getLocation(1);
-		Assert.assertThat(MetadataUtils.getLocation("dc5c1fcc-0459-4201-bf70-0b90535ba362"), is(unknown));
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: Did DbSessionFactory require changing this uuid?

Comment (asp78) [inline api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: Yes it threw errors otherwise

Comment (dkayiwa) [inline api/src/test/java/org/openmrs/module/metadatadeploy/MetadataUtilsTest.java]: And were these errors caused by your change? If yes, how?

**Comment by dkayiwa** [comment]
@asp78 are you still working on this?

**Comment by dkayiwa** [comment]
@asp78 ping me to reopen when you resume working on this.

**Pull request closed without merging**